controllers/openstackcluster_controller.go

```diff
@@ -745,6 +745,7 @@ func reconcileControlPlaneEndpoint(scope *scope.WithLogger, networkingService *n
 	case openStackCluster.Spec.APIServerLoadBalancer.IsEnabled():
 		loadBalancerService, err := loadbalancer.NewService(scope)
 		if err != nil {
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to init load balancer client: %w", err))
```

Comment:
There are a couple of problems with this.

Firstly, not all errors returned here would be terminal errors. I think I would want to be matching specifically on an 'endpoint not found' error here. For example, we wouldn't want to go marking all clusters perma-failed just because of a temporary API outage.

Secondly, we unfortunately don't actually have a perma-failed condition for clusters, although I think we should and this would be an example of when we'd use it. We do have this for OpenStackMachine, but not for OpenStackCluster. Before implementing one we'd need to audit most uses of this function.

Reply:
fine,I will add it

Reply:
Your suggestion is that this change may require reviewing the entire failure handling process of OpenStack Cluster, not just in this one place?

Reply:
Unfortunately, I think so. We'd have to add a 'failed' condition to start with, which we don't currently have, and we'd need to ensure we don't set it anywhere unintentionally.

I think the effort would be worthwhile, though. We should have this.

Reply:
I have got it, I will try my best to complete this PR.

Reply:
How do I handle the compatibility of v1alpha6 and v1alpha7 versions? This is my first time submitting code to this community and I need help in this area. Are there any examples or explanations I can refer to?  @mdbooth